uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorPlugin.java

```diff
@@ -0,0 +1,31 @@
+/*
+* Copyright 2015 JBoss Inc
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*       http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.uberfire.ext.layout.editor.client;
+
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.ext.plugin.model.LayoutEditorModel;
+import org.uberfire.ext.plugin.model.PluginType;
+import org.uberfire.mvp.ParameterizedCommand;
+
+public interface LayoutEditorPlugin extends LayoutEditor {
+
+    void load(PluginType pluginType, Path currentPath, ParameterizedCommand<LayoutEditorModel> loadCallBack);
+
+    void save(Path path, RemoteCallback<Path> saveSuccessCallback);
+
+    int getCurrentModelHash();
```

Comment:
I would remove this method and make perspective editor do: getModel().hashcode()